app/src/common/shared/org/mozilla/vrbrowser/telemetry/GleanMetricsService.java

```diff
@@ -63,23 +63,19 @@ public static void init(@NonNull Context aContext, @NonNull Client client) {
         initialized = true;
 
         final boolean telemetryEnabled = SettingsStore.getInstance(aContext).isTelemetryEnabled();
-        if (telemetryEnabled) {
-            GleanMetricsService.start();
-        } else {
-            GleanMetricsService.stop();
-        }
         Configuration config = new Configuration(
                 ConceptFetchHttpUploader.fromClient(client),
                 Configuration.DEFAULT_TELEMETRY_ENDPOINT,
                 BuildConfig.BUILD_TYPE);
-        Glean.INSTANCE.initialize(aContext, true, config);
+
+        Glean.INSTANCE.initialize(aContext, telemetryEnabled, config);
+        setStartupMetrics();
     }
 
     // It would be called when users turn on/off the setting of telemetry.
     // e.g., SettingsStore.getInstance(context).setTelemetryEnabled();
     public static void start() {
         Glean.INSTANCE.setUploadEnabled(true);
-        setStartupMetrics();
```

Comment:
You should retain this as well: if user toggle telemetry of and then on again, the metrics will be cleared and you'll need to set them again.

Reply:
Yes, you are right. I use Glean testing API can verify my channel name is cleared once users switch off Glean.

Reply:
Cool, let me know once you restore this line so that I will happily r+ :)

Reply:
Restored. :)